File: src/test/java/selenium/tests/MainPage.java
package selenium.tests;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * Created by Administrator on 2016-09-23.
 */
public class MainPage extends Abstract {
    public MainPage(WebDriver driver) {
        super(driver);
    }

    public void goToPostView(String title) {
        driver.findElement(By.xpath("//div[@id='content']//a[contains(text(),'"+title+"')]")).click();
    }
}

File: TestAutomation/pages/MainPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using OpenQA.Selenium;

namespace TestAutomation.pages
{
    class MainPage : Page
    {
        public MainPage(IWebDriver driver) : base(driver)
        {
        }

        internal void openMainPage()
        {
            driver.Navigate().GoToUrl(baseURL);
        }
        internal PostPage openPost(string postURL)
        {
            driver.Navigate().GoToUrl(postURL);
            return new PostPage(driver);
        }

        internal string findPostURL(int postOrdinalNumber)
        {
            return driver.FindElement(By.XPath("//article[" + postOrdinalNumber.ToString() + "]//*[@class='entry-title']//a")).GetAttribute("href");
        }
    }
}
